src/background/toolbarIconHandler.js

```diff
@@ -0,0 +1,59 @@
+/* This Source Code Form is subject to the terms of the Mozilla Public
+ * License, v. 2.0. If a copy of the MPL was not distributed with this
+ * file, You can obtain one at http://mozilla.org/MPL/2.0/. */
+
+import { Component } from "./component.js";
+import { VPNController, VPNState } from "./vpncontroller/index.js";
+
+/**
+ * ToolbarIconHandler updates the browserAction (toolbar) icon 
+ * to reflect the VPN client's status and updates the icon in
+ * response to darkmode/lightmode changes.
+ */
+export class ToolbarIconHandler extends Component {
+  /**
+   *
+   * @param {*} receiver
+   * @param {VPNController} controller
+   */
+  constructor(receiver, controller) {
+    super(receiver);
+    this.controller = controller;
+  }
+
+  /** @type {VPNState | undefined} */
+  controllerState;
+
+  async init() {
+    this.controller.state.subscribe((s) => {
```

Comment:
Let's add a todo to move this from the vpn-client on/off state to the extension on/off state, so we dont forget

Reply:
done!